tests/functional/adapter/test_cratedb.py

```diff
@@ -0,0 +1,104 @@
+import pytest
+from dbt.tests.adapter.simple_snapshot import common
+
+from dbt.tests.util import check_relations_equal, run_dbt
+
+
+basic_sql = """
+SELECT 42
+"""
+
+rename_table_sql = """
+{% set exists, table_source = get_or_create_relation(database=None, schema=target.schema, identifier="table_source", type="table") %}
+{% set exists, table_target = get_or_create_relation(database=None, schema=target.schema, identifier="table_target", type="table") %}
+
+{# Create table `source`. #}
+{% set sql = get_create_table_as_sql(False, table_source, "SELECT 42 as value") %}
+{% do run_query(sql) %}
+
+{# Rename table to `target`. #}
+{% do adapter.rename_relation(table_source, table_target) %}
+
+-- Dummy statement.
+SELECT TRUE AS id
+"""
+
+
+rename_view_sql = """
+{% set exists, view_source = get_or_create_relation(database=None, schema=target.schema, identifier="view_source", type="view") %}
+{% set exists, view_target = get_or_create_relation(database=None, schema=target.schema, identifier="view_target", type="view") %}
+
+{# Create view `source`. #}
+{% set sql = get_create_view_as_sql(view_source, "SELECT 42 as value") %}
+{% do run_query(sql) %}
+
+{# Rename view to `target`. #}
+{% do adapter.rename_relation(view_source, view_target) %}
+
+-- Dummy statement.
+SELECT TRUE AS id
+"""
```

Comment:
Dear @hlcianfagna,

you asked about the `rename_relation` macro over at https://github.com/crate/cratedb-examples/pull/754#pullrequestreview-2465232931. Thank you very much.

Hereby, I am adding an integration test with CrateDB that is using vanilla dbt Core's / dbt-postgres' `rename_relation` macro, and it seems like it works well without much ado.

So, do you think the other `rename_relation` you provided per [^1], originally taken from [^2], will still be neeeded?

With kind regards,
Andreas.


[^1]: https://github.com/crate/cratedb-examples/blob/91c0bb9f74cf62b67a4aabe5a79f7c80b475d16b/framework/dbt/basic/macros/overrides.sql#L50-L86
[^2]: https://community.cratedb.com/t/using-dbt-with-cratedb/1566

Reply:
I think it could be removed now, thank you.